File: paoc/paoc.negocio/src/main/java/br/cefetmg/paoc/negocio/excessoes/negocioException.java
package br.cefetmg.paoc.negocio.excessoes;

public class negocioException extends Exception {
    private int codigo;
    
    public negocioException(String str){
        super(str);
        codigo = 0;
    }
    
    public negocioException(int codigo, String str) {
        super(str);
        this.codigo = codigo;
    }
    
     public int getCodigo() {
        return codigo;
    }
}

File: services/negocioException.php
<?php
class negocioException extends Exception {
    private $codigo;

    public function __construct($str, $codigo = 0) {
        parent::__construct($str);
        $this->codigo = $codigo;
    }

    public function getCodigo() {
        return $this->codigo;
    }
}
?>
